pkg/trace/api/otlp.go

```diff
@@ -54,11 +55,31 @@ type OTLPReceiver struct {
 	out         chan<- *Payload     // the outgoing payload channel
 	conf        *config.AgentConfig // receiver config
 	cidProvider IDProvider          // container ID provider
+
+	// recvsem is a semaphore that controls the number goroutines that can
+	// be simultaneously deserializing incoming payloads.
+	// It is important to control this in order to prevent decoding incoming
+	// payloads faster than we can process them, and buffering them, resulting
+	// in memory limit issues.
+	recvsem chan struct{}
 }
 
 // NewOTLPReceiver returns a new OTLPReceiver which sends any incoming traces down the out channel.
 func NewOTLPReceiver(out chan<- *Payload, cfg *config.AgentConfig) *OTLPReceiver {
-	return &OTLPReceiver{out: out, conf: cfg, cidProvider: NewIDProvider(cfg.ContainerProcRoot)}
+	semcount := cfg.Decoders
+	if semcount == 0 {
+		semcount = runtime.GOMAXPROCS(0) / 2
+		if semcount == 0 {
+			semcount = 1
+		}
+	}
+	log.Infof("OTLPReceiver configured with %d decoders and a timeout of %dms", semcount, cfg.DecoderTimeout)
```

Comment:
Is this that important that it needs to be at the `info` level?

Reply:
I have no strong opinion. We left it at info level in the Datadog receiver because it is new and having this info in the logs could be very helpful for debugging as these changes roll out. Happy to make it debug / whatever.